scipy/linalg/tests/test_batch.py

```diff
@@ -324,3 +324,126 @@ def test_eigh_tridiagonal(self, fun_n_out, dtype, rng):
         d = get_random((3, 4, 5), dtype=dtype, rng=rng)
         e = get_random((3, 4, 4), dtype=dtype, rng=rng)
         self.batch_test(fun, (d, e), core_dim=1, n_out=n_out, broadcast=False)
+
+    @pytest.mark.parametrize('bdim', [(5,), (5, 4), (2, 3, 5, 4)])
+    @pytest.mark.parametrize('dtype', floating)
+    def test_solve(self, bdim, dtype, rng):
+        A = get_random((2, 3, 5, 5), dtype=dtype, rng=rng)
+        b = get_random(bdim, dtype=dtype, rng=rng)
+        x = linalg.solve(A, b)
+        if len(bdim) == 1:
+            x = x[..., np.newaxis]
+            b = b[..., np.newaxis]
+        assert_allclose(A @ x - b, 0, atol=1e-6)
```

Comment:
I guess this checks everything follows standard broadcasting rules but I wonder if here and below we also compare the solution against `np.linalg.solve`/`np.linalg.lstsq` to be extra sure?

Reply:
Ok, I will try to do that where I can easily. Where I've created the matrix in banded form directly, though, I think I'll omit it.

Reply:
Done for half the tests. `np.linalg.lstsq` doesn't broadcast.